src/main/scala/faces/apps/RandomFaces.scala

```diff
@@ -100,27 +100,33 @@ object RandomFaces extends App {
             centered.camera.principalPoint.y + (2.0 * yTranslationDistribution()) / imageHeight)
         ))
 
-        // render image with or without background
-        val img = if(bg) {
-          require(helpers.loadBgs.nonEmpty, "no Background files with type " + cfg.backgrounds.bgType + " found in " + cfg.backgrounds.bgPath)
-          val rndBG = helpers.loadBgs(rnd.scalaRandom.nextInt(helpers.loadBgs.length))
-          val rndBGimg = PixelImageIO.read[RGBA](rndBG).get.resample(imageWidth, imageHeight)
-          helpers.renderer.renderImage(rps).zip(rndBGimg).map(p => if (p._1.a < 0.5) p._2 else p._1 )
-        }
-        else{
-          helpers.renderer.renderImage(rps)
-        }
+        val imageData =
+          for ((postfix, currentRenderer) <- helpers.renderingMethods) yield {
+            if (bg && postfix == "") { // only allow different backgrounds for standard renderings
+              require(helpers.loadBgs.nonEmpty, "no Background files with type " + cfg.backgrounds.bgType + " found in " + cfg.backgrounds.bgPath)
+              val rndBG = helpers.loadBgs(rnd.scalaRandom.nextInt(helpers.loadBgs.length))
+              val rndBGimg = PixelImageIO.read[RGBA](rndBG).get.resample(imageWidth, imageHeight)
+              (currentRenderer.renderImage(centered).zip(rndBGimg).map(p => if (p._1.a < 0.5) p._2 else p._1), postfix)
+            }
+            else {
+              (currentRenderer.renderImage(centered), postfix)
+            }
+          }
 
         // write images and their parameters
         println(s"Generating \t ID:$id \t Sample:$n")
-        helpers.write(img, rps, id, n)
-
-      }}
+        for ((img, postifx) <- imageData) {
+          helpers.writeExceptImage(centered, id, n)
+          helpers.writeImg(img, id, n, postifx)
```

Comment:
Why are here `writeExceptImage` and `writeImg` called? Internally `writeImg` calls `writeExceptImg`. To me this seems that here half of it is done twice.

Reply:
Sorry my mistake.... `write` does both but not `writeImg`.